apis/cassandra/v1beta1/cassandradatacenter_types.go

```diff
@@ -747,6 +748,28 @@ func (dc *CassandraDatacenter) GetContainerPorts() ([]corev1.ContainerPort, erro
 	return ports, nil
 }
 
+func (dc *CassandraDatacenter) FullQueryEnabled() (bool, error) {
```

Comment:
You can't stub methods like that in any case, since this isn't an interface. I don't follow the enabled part of k8ssandra, this isn't a Helm chart.

Reply:
If you really had to stub it you could pass the `FullQueryEnabled()`function as a parameter to the caller. You can't even do that though if it is a method of `CassandraDatacenter`. I'd 100% support you making `CassandraDatacenter` an interface though, it would be quite helpful in testing 😉

Reply:
A struct shouldn't be an interface. Never. I think that approach is looking Go from wrong direction, this isn't Java and its reflection capabilities. 

This is a method that works as a method for CassandraDatacenter and should never be stubbed, if you have to do that, then you're doing something wrong. This is very stateful one.

Reply:
> A struct shouldn't be an interface. Never. I think that approach is looking Go from wrong direction, this isn't Java and its reflection capabilities.

So I'm not proposing that the struct be an interface. I'm proposing to embed an interface `FQLthingDoer` in `CassandraDatacenter` and attaching FQL functions to it. 

I'm not sure where reflection comes into this, and I steer people away from using reflection in JVM languages because it is almost always bad practice IMO.

_* NB: `FQLFunctionDoer` is clearly a placeholder name and we should come up with something better, please not `FQLManager` because I hate that Javaism too_ 

> This is a method that works as a method for CassandraDatacenter and should never be stubbed, if you have to do that, then you're doing something wrong. This is very stateful one.

I don't understand what statefulness has to do with stubbing methods. I think we disagree on how to construct unit tests. I'm not one to be dogmatic on this, so I'll resolve this comment, but I just want to make sure I'm expressing my point clearly.